OPM/opm-rule-onboarding.ps1

```diff
@@ -0,0 +1,540 @@
+		exit
+	}
+#endregion
+
+#region [Read Accounts CSV file and Create Accounts]
+	If([string]::IsNullOrEmpty($CsvPath))
+	{
+		$CsvPath = OpenFile-Dialog($g_CsvDefaultPath)
+	}
+	$delimiter = $(If ($CsvDelimiter -eq "Comma") { "," } else { "`t" } )
+	$rulesCSV = Import-CSV $csvPath -Delimiter $delimiter
+	$rowCount = $($rulesCSV.Safe.Count)
+	$counter = 0
+	$successCounter = 0
+	Log-Msg -Type Info -MSG "Starting to Onboard $rowCount rules." -SubHeader
+	ForEach ($rule in $rulesCSV)
+	{
+		if ($null -ne $rule)
+		{
+			try {
+				# Check mandatory fields
+				If([string]::IsNullOrEmpty($rule.PlatformId)) { throw "Missing mandatory field: Platform ID" }
+				If([string]::IsNullOrEmpty($rule.Command)) { throw "Missing mandatory field: Command" }
+				If([string]::IsNullOrEmpty($rule.UserName)) { throw "Missing mandatory field: UserName" }
+				
+				$rule.PlatformId = (Get-TrimmedString $rule.PlatformID)
+				$objRule = "" | Select-Object "Command", "CommandGroup", "PermissionType", "Restrictions", "UserName"
+				$objRule.Command = (Get-TrimmedString $rule.Command)
+				$objRule.CommandGroup = (Get-TrimmedString $rule.CommandGroup)
+				$objRule.PermissionType = (Get-TrimmedString $rule.PermissionType)
+				$objRule.Restrictions = (Get-TrimmedString $rule.Restrictions)
+				$objRule.UserName = (Get-TrimmedString $rule.UserName)
+				
+				if ([string]::IsNullOrEmpty($objRule.CommandGroup)) { 
+					$objRule.CommandGroup = $false
+				} elseif ($objRule.CommandGroup.ToLower() -eq "yes" -or $objRule.CommandGroup.ToLower() -eq "true") {
+					$objRule.CommandGroup = $true
+				} else {
+					$objRule.CommandGroup = $false
+				}
+				
+				if ([string]::IsNullOrEmpty($objRule.PermissionType)) { 
+					$objRule.PermissionType = "Allow"
+				} elseif ($objRule.permissionType.ToLower() -eq "allow" -or $objRule.permissionType.ToLower() -eq "deny") {
+					$objRule.PermissionType = (Get-Culture).TextInfo.ToTitleCase($objRule.PermissionType)
+				} else {
+					throw "Invalid field: PermissionType"
+				}
+				
+				if ($objRule.commandGroup -eq $false -and -Not $objRule.command.StartsWith("/") -and -Not $objRule.command -eq ".*") {
+					Write-Warning "Rule Number $counter : It is a recommended security best practice to define commands with an absolute path. Please reconfigure rules like (cat /etc/passwd) to (/usr/bin/cat /etc/passwd)" -WarningAction Inquire
+				}
+				try {
+					# Create the rule
+					$restBody = $objRule | ConvertTo-Json -Depth 5
+					$URL_PlatformCommands = $URL_PlatformCommands -f $rule.platformID
+					Log-Msg -Type Debug -Msg $restBody
+					$addRuleResult = $(Invoke-RestMethod -Method Put -Uri $URL_PlatformCommands -Headers $g_LogonHeader -Body $restBody -ContentType "application/json" -TimeoutSec 60)
+					if($addRuleResult -ne $null) {
+						Log-Msg -Type Info -MSG "Rule Onboarded Successfully"
+						# Increment counter
+						$successCounter++
+						Log-Msg -Type Info -MSG "[$counter/$rowCount] Added successfully."  
```

Comment:
Is this Counter or SuccessCounter?

Reply:
It sort of depends on what your objective is. 
In my mind, I added to your code here with the ability to:
- Differentiate between a line in the input (for tracking input errors) and the successful addition of that input to the vault
- Display which input row was successfully added while also tracking how many were successful. 
I'll add some language to clarify.